wallet-new/src/Cardano/Wallet/API/V1/Migration.hs

```diff
@@ -108,3 +111,60 @@ instance Migrate V0.SyncProgress V1.SyncProgress where
                 Just nd | _spLocalCD >= nd -> 100
                 Just nd -> round @Double $ (fromIntegral _spLocalCD / fromIntegral nd) * 100.0
         in pure $ V1.mkSyncProgress (fromIntegral percentage)
+
+
+--
+instance Migrate V0.CAccount V1.Account where
+    eitherMigrate V0.CAccount{..} =
+        V1.Account <$> eitherMigrate caId
+                  -- ^ accId
+                  <*> mapM eitherMigrate caAddresses
+                  -- ^ accAddresses
+                  <*> eitherMigrate caAmount
+                  -- ^ accAmount
+                  <*> pure (V0.caName caMeta)
+                  -- ^ accName
+                  <*> eitherMigrate caId
+                  -- ^ accWalletId
+
+instance Migrate V1.WalletId (V0.CId V0.Wal) where
+    eitherMigrate (V1.WalletId h) = pure $ V0.CId (V0.CHash h)
+
+--
+-- Following instances are friendly borrowed from @martoon's PR https://github.com/input-output-hk/cardano-sl/pull/2008
+-- TODO (jk): Use instances of his PR when it has been merged
+
+-- in old API 'V0.AccountId' supposed to carry both wallet id and derivation index
+instance Migrate (V1.WalletId, V1.AccountId) V0.AccountId where
+    eitherMigrate (walId, accId) =
+        V0.AccountId <$> eitherMigrate walId <*> pure accId
+
+instance Migrate V0.AccountId (V1.WalletId, V1.AccountId) where
+    eitherMigrate accId =
+        (,) <$> eitherMigrate (V0.aiWId accId) <*> pure (V0.aiIndex accId)
+
+instance Migrate V0.CAccountId V0.AccountId where
+    eitherMigrate = first Errors.MigrationFailed . decodeCType
+
+--
+-- #end of TODO (jk) ^
+--
+
+instance Migrate V0.CAccountId V1.AccountId where
+    eitherMigrate cAccId = do
+        -- V0.CAccountId -> V0.AccountId -> V1.AccountId
+        a :: V0.AccountId <- eitherMigrate cAccId
+        b :: (V1.WalletId, V1.AccountId) <- eitherMigrate a
+        pure $ snd b
```

Comment:
I find this function a bit dense. If we need to add type signatures under the form of comments we can probably do better. What about something along the lines of:

``` haskell
eitherMigrate clientAccountId = do
  oldAccountId <- eitherMigrate clientAccountId
  (walletId, newAccountId) <- eitherMigrate oldAccountId
  pure newAccountId
```

Reply:
Oh yessss, naming variables like `a` or `b` make no sense here. I will change it. What we still need here are type annotations to guide `ghc` and to not get errors such as `Ambiguous type variables ...`. As always I'm open for better solutions, of course :)

Reply:
> As always I'm open for better solutions, of course :)

@sectore Do you happen to like `TypeApplications`? Usually you need just one in the right place to allow GHC to infer the whole thing -- did you give it a whirl? 😉